src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -59,7 +64,27 @@ public SchedulerClient build() {
 		public void scheduleForExecution(Instant exeecutionTime,
 				TaskInstance taskInstance) {
 			taskRepository.createIfNotExists(new Execution(exeecutionTime, taskInstance));
-		} 
+		}
+
+		@Override
+		public void reschedule(String taskName, String instanceId, Instant newExecutionTime) {
+			Optional<Execution> execution = taskRepository.getExecution(taskName, instanceId);
+			if(execution.isPresent()) {
+				taskRepository.reschedule(execution.get(), newExecutionTime, null, null);
+			} else {
+				throw new RuntimeException(String.format("Could not reschedule - no task with name '%' and id '%s' was found." , taskName, instanceId));
+			}
+		}
+
+		@Override
+		public void cancelSchedule(String taskName, String instanceId) {
+			Optional<Execution> execution = taskRepository.getExecution(taskName, instanceId);
+			if(execution.isPresent()) {
+				taskRepository.remove(execution.get());
```

Comment:
if `picked = true` here, throw exception about not allowed to cancel executions that are currently executing